public/docs/_examples/server-communication/dart/lib/wiki/wikipedia_service.dart

```diff
@@ -0,0 +1,24 @@
+// #docregion
+import 'dart:async';
+import 'package:angular2/angular2.dart';
+import 'package:jsonpadding/jsonpadding.dart';
+
+@Injectable()
+class WikipediaService {
+  Future<List<String>> search(String term) async {
+    // #docregion call-jsonp
+    Uri uri = new Uri(
+        scheme: 'http',
+        host: 'en.wikipedia.org',
+        path: 'w/api.php',
+        queryParameters: {
+          'search': term,
+          'action': 'opensearch',
+          'format': 'json'
+        });
+    // TODO: Error handling
+    List result = await jsonp(uri);
```

Comment:
The API of jsonp is a bit awkward here, it's not immediately clear why we need to return result[1].